src/Shared/Logger/infra/logger.go

```diff
@@ -1,137 +1,62 @@
-		StopChan:    make(chan bool),
-		Done:        make(chan bool),
-	}
-}
-
-func createLogDir(path string) {
-	os.MkdirAll(path, 0777)
-}
-
-func (logger *Logger) AddEntry(entry domain.Entry) {
-	logger.EntryChan <- entry
-}
-
-func (logger *Logger) addEntryToBuffer(entry domain.Entry) {
-	buffer := logger.getBuffer(entry)
-	bytesBuf, err := io.ReadAll(entry.Value)
-	if err != nil {
-		log.Fatalf("error reading entry reader: %v", err)
-	}
-
-	*buffer = append(*buffer, string(bytesBuf))
-	logger.buffers[entry.Id] = buffer
-}
-
-func (logger *Logger) getBuffer(entry domain.Entry) *[]string {
-	buffer, exists := logger.buffers[entry.Id]
-
-	if !exists {
-		logger.buffers[entry.Id] = new([]string)
-		buffer = logger.buffers[entry.Id]
-	}
-
-	return buffer
-}
-
-func (logger *Logger) write() {
-	for name := range logger.buffers {
-		logger.writeBufferToFile(name)
-	}
-	logger.clearBuffers()
-}
-
-func (logger *Logger) writeBufferToFile(bufferName string) {
-	file := logger.getFile(bufferName)
-	for _, text := range *logger.buffers[bufferName] {
-		file.WriteString(fmt.Sprintf("%v\n", text))
-	}
-}
-
-func (logger *Logger) getFile(fileName string) *os.File {
-	file, exists := logger.files[fileName]
-	if !exists {
-		file = createFile(logger.baseDir, fileName)
-	}
-
-	return file
-}
-
-func createFile(dirPath string, fileName string) *os.File {
-	file, err := os.Create(path.Join(dirPath, fileName+".txt"))
-	if err != nil {
-		log.Fatalf("write values: %s\n", err)
-	}
-
-	return file
-}
-
-func (logger *Logger) clearBuffers() {
-	logger.buffers = make(map[string]*[]string)
-}
-
-func (logger *Logger) Stop() {
-	logger.StopChan <- true
-	<-logger.Done
-}
-
-func (logger *Logger) close() {
-	for _, file := range logger.files {
-		file.Close()
-	}
-	logger.writeTicker.Stop()
-	logger.Done <- true
-}
-
-func (logger *Logger) Record() {
-	go func() {
-	loop:
-		for {
-			select {
-			case <-logger.writeTicker.C:
-				logger.write()
-			case entry := <-logger.EntryChan:
-				logger.addEntryToBuffer(entry)
-			case <-logger.StopChan:
-				logger.write()
-				logger.close()
-				break loop
-			}
-		}
-	}()
-}
+package infra
+
+import (
+	"path"
+	"time"
+
+	"github.com/HyperloopUPV-H8/Backend-H8/DataTransfer/domain"
+)
+
+type Logger struct {
+	baseDir    string
+	currDir    *Dir
+	ticker     *time.Ticker
+	EnableChan chan bool
+	EntryChan  chan domain.PacketTimestampPair
+}
+
+func NewLogger(baseDir string, delay time.Duration) Logger {
+	return Logger{
+		baseDir:    baseDir,
+		currDir:    nil,
+		ticker:     time.NewTicker(delay),
+		EnableChan: make(chan bool),
+		EntryChan:  make(chan domain.PacketTimestampPair, 100),
+	}
+}
+
+func (log Logger) Run() {
+	go func() {
+		for {
+			isEnable := <-log.EnableChan
+			if isEnable {
+				log.record()
+			}
+		}
+	}()
+}
+
+func (log Logger) record() {
+	log.currDir = NewDir(path.Join(log.baseDir, time.Now().Format("2006-01-02_15-04-05")))
+loop:
+	for {
+		select {
+		case <-log.ticker.C:
+			log.currDir.Dump()

```

Comment:
what does "lod.currDir.Dump" do?

Reply:
Each file has a corresponding array that fills up with entries, and every once in a while the content from these arrays is copied into the corresponding file. Dump puts the contents of each array in the corresponding file.